src/Apache.IoTDB/SessionPool.cs

```diff
@@ -101,12 +175,86 @@ public async Task Open(CancellationToken cancellationToken = default)
         {
             _clients = new ConcurrentClientQueue();
             _clients.Timeout = _timeout * 5;
-            for (var index = 0; index < _poolSize; index++)
+            if (_nodeUrls.Count == 0)
+            {
+                for (var index = 0; index < _poolSize; index++)
+                {
+                    _clients.Add(await CreateAndOpen(_host, _port, _enableRpcCompression, _timeout, cancellationToken));
+                }
+            }
+            else
+            {
+                foreach (var endPoint in _endPoints)
+                {
```

Comment:
Here, we may choose different nodes to initialize the pool

Reply:
Are we assuming here that all nodes are available? If a node is unavailable, should we throw an error or connect to other nodes?

Reply:
The definition of reconnection is to reconnect to the next available node.
maybe see  go session or java session